### PR TITLE
samples: usb: remove unnecessary GPIO dependency

### LINKED ISSUE #29034: error in samples/subsys/usb/cdc_acm
**Describe the bug**
The CDC ACM sample isn't usable on native_posix. It pulls needlessly GPIO drivers in, without using them.

**To Reproduce**
1. cd samples/subsys/usb/cdc_acm
2. west build --pristine -b native_posix
3. Error at compilation time:

```
-- Configuring done
CMake Error at ../../../../cmake/extensions.cmake:398 (add_library):
  No SOURCES given to target: drivers__gpio
```

**Expected behavior**
I expected that the compilation actually worked, as USB seems to be emulated when compiled for native_posix and can be used via the Linux "usbip" command line tool.

BTW, removing the offending line fixes the issue:

1. sed -i '/CONFIG_GPIO/d' prj.conf 
2. west build --pristine -b native_posix

**Impact**

Unable to test the sample in native posix mode.

**Environment (please complete the following information):**
 - OS: Linux
 - git describe --tags: zephyr-v2.4.0-388-g5bda586c64  (which is git HEAD as of now)

### PR DESCRIPTION
**Pull request by jfischer-no**
Remove unnecessary GPIO dependency.

Fixes: #29034

### CODE REVIEW

**Review by eobalski — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

File: samples/subsys/usb/cdc_acm/sample.yaml [hunk 1]
```diff
@@ -12,9 +13,15 @@ tests:
...
     extra_args: "-DOVERLAY_CONFIG=overlay-composite-cdc-msc.conf"
     harness: console
     harness_config:
       type: one_line
       regex:
         - "Wait for DTR"
+  sample.usb.cdc-acm.buildonly:
+    depends_on: usb_device
+    tags: usb
+    platform_allow: native_posix native_posix_64
```
Comment (galak) [inline samples/subsys/usb/cdc_acm/sample.yaml:26]: Do we need the platform_allow?

**Review by galak — DISMISSED** [review]
I think the `platform_allow:` for the buildonly test should be `integration_platforms:`

File: samples/subsys/usb/cdc_acm/sample.yaml (hunk 1 above)
Comment (jfischer-phytec-iot) [reply samples/subsys/usb/cdc_acm/sample.yaml:26]: I think so, for the native_posix samples should only be built and not executed in CI.

Comment (jfischer-phytec-iot) [reply samples/subsys/usb/cdc_acm/sample.yaml:26]: @galak ping

**Review by carlescufi — APPROVED** [review]

**Pull request merged**